game/state/shared/aequipment.cpp

```diff
@@ -702,12 +702,18 @@ void AEquipment::fire(GameState &state, Vec3<float> targetPosition, StateRef<Bat
 	auto unit = ownerAgent->unit;
 	auto payload = getPayloadType();
 	Vec3<float> originalTarget = targetPosition;
+	char number_of_shots;
```

Comment:
I'm not a big fan of 'char' data types (for anything not a string character)

If you don't really care about the size, use 'int'.

If you do, use one of the stdint.h types (int8_t or whatever)

Reply:
No problem.